src/util/simplify_expr_int.cpp

```diff
@@ -42,7 +42,10 @@ bool simplify_exprt::simplify_bswap(bswap_exprt &expr)
     mp_integer new_value=0;
     for(std::size_t bit = 0; bit < width; bit += bits_per_byte)
     {
-      assert(!bytes.empty());
+      INVARIANT(
+        !bytes.empty(),
+        "bytes is not empty because we just pushed just as many elements on "
+        "top of it as we are popping now");
```

Comment:
@tautschnig I'm inclined to feel that the invariant here is checking for UB in case someone makes an error in the loop condition.  
Moving the invariant out of the loop would seem to check something that is trivially true (I think...).

Reply:
Ok, not a big deal, I personally would have thought we don't need to guard against all possible future errors, but figuring out where exactly to draw this line is hard. Surely not a blocker here.